ext/dom/node.c

```diff
@@ -98,7 +98,8 @@ int dom_node_node_name_read(dom_object *obj, zval *retval)
 			str = "#text";
 			break;
 		default:
-			php_error_docref(NULL, E_WARNING, "Invalid Node Type");
+			zend_value_error("XML Node type must be one of XML_*_NODE");
```

Comment:
This error message doesn't really make sense, the node type is always one of XML_*_NODE. This should be something like "Cannot get name for this node type", or possibly even an assertion if this is supposed to work for all node types (@beberlei ?)

Reply:
all node types are handled here, this should only fail if libxml adds something new, which is highly unlikely. The user would not be able to address this error as its a C programming error not a PHP one, so the message could better be "Unknown node type in libxml datastructure that DOM extension cannot handle." - I don't expect it to be ever seen by anyone

Reply:
I'll add this as an Assertion then